ansible_collections/arista/avd/roles/eos_designs/python_modules/core_interfaces_and_l3_edge/utils.py

```diff
@@ -172,7 +173,8 @@ def _get_p2p_data(self, p2p_link: dict) -> dict:
                 var_name=f"{peer} under {self.data_model}.p2p_links.[].port_channel.nodes_child_interfaces",
             )
             peer_member_interfaces = peer["interfaces"]
-            peer_id = peer.get("channel_id", int("".join(re.findall(r"\d", peer_member_interfaces[0]))))
+            peer_inteface_digits = re.findall(r"\d", peer_member_interfaces[0])
+            peer_id = peer.get("channel_id", int("".join(peer_inteface_digits)))
```

Comment:
IMO This is even more confusing :-)

The int/join/findall construct is used many places in AVD, so here we just split the default value out as we do many other places.
```suggesting
            default_peer_channel_id = int("".join(re.findall(r"\d", peer_member_interfaces[0])))
            peer_id = peer.get("channel_id", default_peer_channel_id)
```

Reply:
Updated the logic here as requested.